src/transports/local/pipelines.ts

```diff
@@ -0,0 +1,97 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * /!\ This file is feature-switchable.
+ * It always should be imported through the `features` object.
+ */
+
+import { of as observableOf } from "rxjs";
+import Manifest from "../../manifest";
+import parseLocalManifest, {
+  ILocalManifest,
+} from "../../parsers/manifest/local";
+import {
+  IManifestLoaderArguments,
+  IManifestLoaderObservable,
+  IManifestParserArguments,
+  IManifestParserObservable,
+  ITransportOptions,
+  ITransportPipelines,
+} from "../types";
+import callCustomManifestLoader from "../utils/call_custom_manifest_loader";
+import segmentLoader from "./segment_loader";
+import segmentParser from "./segment_parser";
+import textTrackParser from "./text_parser";
+
+/**
+ * Returns pipelines used for local Manifest streaming.
+ * @param {Object} options
+ * @returns {Object}
+ */
+export default function getLocalManifestPipelines(
+  options : ITransportOptions
+) : ITransportPipelines {
+
+  const customManifestLoader = options.manifestLoader;
+  const manifestPipeline = {
+    loader(args : IManifestLoaderArguments) : IManifestLoaderObservable {
+      if (customManifestLoader == null) {
+        throw new Error("A local Manifest is not loadable through regular HTTP(S) " +
+                        " calls. You have to set a `manifestLoader` when calling " +
+                        "`loadVideo`");
+      }
+      return callCustomManifestLoader(
+        customManifestLoader,
+        () : never => {
+          throw new Error("Cannot fallback from the `manifestLoader` of a " +
+                          "`local` transport");
+        })(args);
+    },
+
+    parser(
+      { response } : IManifestParserArguments
+    ) : IManifestParserObservable {
+      const manifestData = response.responseData;
+      if (typeof manifestData !== "object") {
+        throw new Error("Wrong format for the manifest data");
+      }
+      const parsed = parseLocalManifest(response.responseData as ILocalManifest);
+      const manifest = new Manifest(parsed, options);
+      return observableOf({ manifest, url: undefined });
+    },
+  };
+
+  const segmentPipeline = { loader: segmentLoader,
+                            parser: segmentParser };
+  const textTrackPipeline = { loader: segmentLoader,
+                              parser: textTrackParser };
+
+  const imageTrackPipeline = {
+    loader:  () : never => {
+      throw new Error("Images track not supported in local transport.");
```

Comment:
Shouldn't we juste log ?

Reply:
Why?
Someone asks for an image but we cannot load it. Sounds like an error to me.